### PR TITLE
Add profile update form

### PR DESCRIPTION
**Pull request by zachgibb**
screen shot 2018-11-09 at 12 47 25 pm
screen shot 2018-11-09 at 12 47 43 pm
screen shot 2018-11-09 at 12 48 32 pm

### CODE REVIEW

File: src/screens/Settings/Profile/Profile.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+      ...this.state.answers,
+      [fieldName]: text,
+    };
+
+    this.setState({answers}, this.checkSavedState);
+  };
+
+  makeSaveButton = () => (
+    <TouchableOpacity style={{flex: 1, justifyContent: 'flex-start'}} onPress={this.handleFormSubmit} hitSlop={calculateHitSlop(20)}>
+      <T.Light style={{color: 'white', marginVertical: 2}}>SAVE</T.Light>
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: lang entry should go here

File: src/screens/Settings/Profile/Profile.js [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+  render() {
+    const { answers } = this.state;
+    const placeholderTextColor = 'rgba(255,255,255,0.75)';
+
+    return (
+      <Layout preload={false} keyboard>
+        <Body scrollable style={styles.body}>
+          <Header>
+            <T.Heading style={styles.headingText}>Profile</T.Heading>
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: lang entry should go here

File: src/screens/Settings/Profile/Profile.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+    const lastNameChanged = oldUser.last_name !== newUser.last_name;
+    const phoneNumberChanged = oldUser.phone_number !== newUser.phone_number;
+    return firstNameChanged || lastNameChanged || phoneNumberChanged;
+  }
+
+  safeFocus = memoize(element => () => invoke(element, 'inputRef.focus'));
+
+  render() {
+    const { answers } = this.state;
+    const placeholderTextColor = 'rgba(255,255,255,0.75)';
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: do you know if this color on the blue background is a11y compliant?

File: src/screens/Settings/Profile/Profile.js [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+                  {this.state.errorMessage}
+                </T.Light>
+              </View>
+            </Try>
+            <Input
+              ref={el => (this.signupFirstNameInput = el)}
+              autoCapitalize="none"
+              autoCorrect={false}
+              editable={!this.state.loading}
+              placeholder="First Name"
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: lang entry should go here

File: src/screens/Settings/Profile/Profile.js [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+              value={answers.first_name}
+              type="underline"
+              style={styles.input}
+            />
+            <Input
+              ref={el => (this.signupLastNameInput = el)}
+              autoCapitalize="none"
+              autoCorrect={false}
+              editable={!this.state.loading}
+              placeholder="Last Name"
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: lang entry should go here

File: src/screens/Settings/Profile/Profile.js [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+              value={answers.last_name}
+              type="underline"
+              style={styles.input}
+            />
+            <Input
+              ref={el => (this.signupUsernameInput = el)}
+              autoCapitalize="none"
+              autoCorrect={false}
+              editable={false}
+              placeholder="Username"
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: lang entry should go here

File: src/screens/Settings/Profile/Profile.js [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+              value={this.props.user.username}
+              type="underline"
+              style={styles.input}
+            />
+            <Input
+              ref={el => (this.signupEmailAddressInput = el)}
+              autoCapitalize="none"
+              autoCorrect={false}
+              editable={false}
+              placeholder="Email"
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: lang entry should go here

File: src/screens/Settings/Profile/Profile.js [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+              value={this.props.user.email_address}
+              type="underline"
+              style={styles.input}
+            />
+            <Input
+              ref={el => (this.signupPhoneInput = el)}
+              autoCapitalize="none"
+              autoCorrect={false}
+              editable={!this.state.loading}
+              placeholder="Phone Number"
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: lang entry should go here

File: src/screens/Settings/Settings/Settings.js [hunk 9]
```diff
@@ -1,29 +1,34 @@
...
   return (
     <Layout preload={false}>
       <Body scrollable style={styles.content}>
         <Link
           title="Seed Words"
           to="SeedWords"
         />
+        {props.isSignedIn && (
+          <Link
+            title="Profile"
```
Comment (ericgibb) [inline src/screens/Settings/Settings/Settings.js]: lang entry should go here

File: src/screens/Settings/index.js [hunk 10]
```diff
@@ -38,6 +39,16 @@ const RoutingStack = createStackNavigator(
           rightAction: null
         }),
     },
+    Profile: {
+      screen: Profile,
+      navigationOptions: navProps =>
+        getNavigationOptions({
+          ...navProps,
+          leftAction: 'back',
+          title: 'Profile',
```
Comment (ericgibb) [inline src/screens/Settings/index.js]: lang entry should go here

File: src/screens/Settings/Profile/Profile.js (hunk 3 above)
Comment (zachgibb) [reply src/screens/Settings/Profile/Profile.js]: how would i check that?

File: src/screens/Settings/Profile/Profile.js [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+    const answers = {
+      ...this.state.answers,
+      [fieldName]: text,
+    };
+
+    this.setState({answers}, this.checkSavedState);
+  };
+
+  makeSaveButton = () => (
+    <TouchableOpacity style={{flex: 1, justifyContent: 'flex-start'}} onPress={this.handleFormSubmit} hitSlop={calculateHitSlop(20)}>
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: can we use `import Button from 'components/Button` here?

File: src/screens/Settings/Profile/Profile.js [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+    const newUser = {
+      username: this.props.user.username,
+      email_address: this.props.user.email_address,
+      first_name: this.state.answers.first_name,
+      last_name: this.state.answers.last_name,
+      phone_number: this.state.answers.phone_number,
+    };
+
+    this.setState({ loading: true }, () => {
+      this.props.navigation.setParams({rightAction: (
```
Comment (ericgibb) [inline src/screens/Settings/Profile/Profile.js]: can this happen before `setState`?

Comment (zachgibb) [reply src/screens/Settings/Profile/Profile.js]: fixed

File: src/screens/Settings/Profile/Profile.js (hunk 11 above)
Comment (zachgibb) [reply src/screens/Settings/Profile/Profile.js]: fixed

**Pull request merged**